src/test/java/net/datafaker/providers/base/PhoneNumberTest.java

```diff
@@ -57,103 +62,139 @@ void testPhone_CA() {
+                    }
                     errorCount++;
                 }
-            } catch (Exception e) {
+            } catch (NumberParseException e) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s (caused by: %s)".formatted(phoneNumber, e));
+                }
                 errorCount++;
             }
         }
-        assertThat(errorCount).isLessThanOrEqualTo(15);
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
     }
 
     @ParameterizedTest
     @MethodSource("generateLanguageAndRegionOfLocales")
-    void testAllPhoneNumberInternational(Locale locale, String phoneNumberRegion) throws NumberParseException {
-        final PhoneNumberUtil util = PhoneNumberUtil.getInstance();
+    void testAllPhoneNumberInternational(Locale locale, String phoneNumberRegion, int allowedErrorsCount) throws NumberParseException {
         int errorCount = 0;
         final BaseFaker faker = new BaseFaker(locale);
         final PhoneNumber phoneNumberProvider = faker.phoneNumber();
-        for (int i = 0; i < 10; i++) {
+        for (int i = 0; i < COUNT; i++) {
             String phoneNumber = phoneNumberProvider.phoneNumberInternational();
             Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
             if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s".formatted(phoneNumber));
+                }
                 errorCount++;
             }
         }
-        assertThat(errorCount).isLessThan(25);
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
+    }
+
+    @ParameterizedTest
+    @MethodSource("generateLanguageAndRegionOfLocales")
+    void testAllPhoneNumberMobile(Locale locale, String phoneNumberRegion, int allowedErrorsCount) throws NumberParseException {
+        int errorCount = 0;
+        final BaseFaker faker = new BaseFaker(locale);
+        final PhoneNumber phoneNumberProvider = faker.phoneNumber();
+        for (int i = 0; i < COUNT; i++) {
+            String phoneNumber = phoneNumberProvider.cellPhone();
+            Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
+            if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s".formatted(phoneNumber));
+                }
+                errorCount++;
+            }
+        }
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
+    }
+
+    private static Arguments args(Locale locale, String phoneNumberRegion) {
+        return args(locale, phoneNumberRegion, (int) (COUNT * 0.77));
+    }
+
+    private static Arguments args(Locale locale, String phoneNumberRegion, int allowedErrorsCount) {
+        return Arguments.of(locale, phoneNumberRegion, allowedErrorsCount);
```

Comment:
should we just `import static` for `Arguments` and then it could be shorter like
```java
of(locale, phoneNumberRegion, allowedErrorsCount);
```
?

Reply:
It's shorter, but is it really readable?
When I read just "of(xxx)", how can I understand what this "of" means"? Of what?

Compared to that, `Arguments.of` is more readable.

Reply:
 I guess it depends on the context
if we are talking about `of` in general then I would agree it is questionable, however here we are in test class, in arguments provider method, with this knowledge it is more readable then from my point of view

Reply:
anyway i'm not going to stop merging just because of this
I expressed my concerns and leave the final decision up to you

Reply:
I'm a huge fan of static imports (AssertJ for example), but in this case, importing "of" doesn't really add to readability, so I'd lean towards keeping the classname there

Reply:
Oh, I'm too late it seems :)